### PR TITLE
Fix below freezing temperature for Inkbird sensors

### LINKED ISSUE #2531: Inkbird IBS-TH1 Temperature is 655.35 Degrees to High When the Temperature is Below Freezing
### The problem

The temperature field for these devices is a 2 byte unsigned integer so when the temperature drops below 0 (in Celsius) the value rolls back around to 65,535 (divided by 100.0f = 655.35) and then drops from there.

### Which version of ESPHome has the issue?

2021.9.2

### What type of installation are you using?

pip

### Which version of Home Assistant has the issue?

2021.10.1

### What platform are you using?

ESP32

### Board

wemos_d1_mini32

### Component causing the issue

inkbird_ibsth1_mini

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by nuttytree**
# What does this implement/fix? 

This fixes the incorrect temperature values for Inkbird sensors when the temperature drops below freezing.

## Types of changes

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes esphome/issues#2531

## Test Environment

- [X] ESP32
- [ ] ESP8266

## Checklist:
  - [X] The code change is tested and works locally.
  - [N/A] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [N/A] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/inkbird_ibsth1_mini/inkbird_ibsth1_mini.cpp [hunk 1]
```diff
@@ -73,6 +73,10 @@ bool InkbirdIbstH1Mini::parse_device(const esp32_ble_tracker::ESPBTDevice &devic
 
   // Read bluetooth data into variable
   auto measured_temperature = mnf_data.uuid.get_uuid().uuid.uuid16 / 100.0f;
+  // The temperature rolls back around to 65535 / 100.0f = 655.35 when it reaches zero so adjust accordingly
+  if (measured_temperature > 327.68) {
+    measured_temperature = measured_temperature - 655.35;
+  }
```
Comment (OttoWinter) [inline esphome/components/inkbird_ibsth1_mini/inkbird_ibsth1_mini.cpp]:
```suggestion
  auto measured_temperature = ((int16_t) mnf_data.uuid.get_uuid().uuid.uuid16) / 100.0f;
```

That looks like it's just 16bit 2's complement, so this should work too (and doesn't depend on floating point math)

**Comment by fkirill** [comment]
Seems like a duplicate for https://github.com/esphome/issues/issues/2531

File: esphome/components/inkbird_ibsth1_mini/inkbird_ibsth1_mini.cpp (hunk 1 above)
Comment (fkirill) [reply esphome/components/inkbird_ibsth1_mini/inkbird_ibsth1_mini.cpp]: yes, it's a type change, see https://github.com/esphome/esphome/pull/2470, seems like we did duplicate fixes. :-)

**Review by OttoWinter — APPROVED** [review]
Thanks

**Pull request merged**